mesa/time.py

```diff
@@ -150,15 +177,25 @@ def step(self) -> None:
 
 
 class StagedActivation(BaseScheduler):
-    """A scheduler which allows agent activation to be divided into several
-    stages instead of a single `step` method. All agents execute one stage
+    """
+    A scheduler allowing agent activation to be divided into several stages, with all agents executing one stage
     before moving on to the next.
 
-    Agents must have all the stage methods implemented. Stage methods take a
-    model object as their only argument.
+    This scheduler is useful for complex models where actions need to be broken down into distinct phases
+    for each agent in each time step. Agents must implement methods for each defined stage.
```

Comment:
May add that this class is a generalization of `SimultaneousActivation`.

Reply:
Done!